File: Data/ChannelRelations.py
import os
import pandas as p

recommends = p.read_csv('.\\2.Analysis\\recommends.csv', delimiter=',')
channels = p.read_json('.\\2.Analysis\\channels.json')


channels = channels[['id','title','subCount']].query('subCount > 10000')
channels.sort_values('subCount', ascending=False)

# save to csv of relations
recommends = recommends.groupby(['FromChannelTitle', 'ChannelTitle', 'FromChannelId', 'ChannelId', 'DistanceFromSeed']).size().to_frame('Size')
chIds = channels['id']
recommends[recommends['FromChannelId'].isin(chIds)].count()
recommends

channels.to_csv('.\\3.Vis\\Channels.csv')
recommends.to_csv('.\\3.Vis\\ChannelRelations.csv', header=True)

File: Data/ChannelRelations.R
library("dplyr", lib.loc="~/R/win-library/3.5")

inDir <- "C:\\Users\\mark\\AppData\\Local\\YoutubeNetworks\\Data\\2018-11-03_11-36\\"
outDir <- "C:\\Users\\mark\\Repos\\YouTubeNetworks\\Site\\static\\data"

setwd(inDir)
channels <- read.csv("Channels.csv", header = TRUE)
outChannels <- channels %>%
  filter(Status=="Seed") %>% 
  select(Id, Title, Status, SubCount, ChannelVideoViews, Type, LR) %>%
  arrange(desc(ChannelVideoViews))



recommends <- read.csv("Visits.csv", header = TRUE)

channelVideos <- recommends %>%
  group_by(FromChannelId) %>%
  summarise(ChannelRecommends = n())

outRelations <- recommends %>%
  group_by(FromChannelTitle, ChannelTitle, FromChannelId, ChannelId) %>%
  subset(ChannelId %in% outChannels$Id) %>% 
  subset(FromChannelId %in% outChannels$Id) %>%
  inner_join(channelVideos, by = "FromChannelId") %>%
  summarise(Recommends = n(), RecommendedViews = sum(FromVideoViews), RecommendsPerVideo=n()/mean(ChannelRecommends)) %>%
  filter(RecommendsPerVideo >= 0) %>%
  arrange(desc(Recommends))

setwd(outDir)
write.csv(outChannels, "Channels.csv")
write.csv(outRelations,"ChannelRelations.csv")


